gnowsys-ndf/gnowsys_ndf/es_initialize.py

```diff
@@ -0,0 +1,751 @@
+from gnowsys_ndf.ndf.models import *
+from elasticsearch import Elasticsearch
+from bson import json_util
+import os
+import sys
+import json
+reload(sys)  # Reload does the trick!
+sys.setdefaultencoding('UTF8')
+es = Elasticsearch("http://elsearch:changeit@gsearch:9200")
+
+# doc_type = ""
+author_map = {}
+group_map = {}
+author_index = "author_clix"
+index = "clix"
+
+
+def index_docs(all_docs):
+	k = 0
+	for node in all_docs:
+		doc = json.dumps(node,default=json_util.default) #convert mongoDB objectt to a JSON string
+		#doc = json.loads(doc,object_hook=json_util.object_hook) ->get back mongoDB object from JSON string
+		document = json.loads(doc) #convert JSON string to a python dictionary
+		#doc = json.dumps(document) #convert python dictionary to JSON string
+		#print(document)
+		# print(document["_type"])
+		if("name" in document.keys()):
+			document["id"] = document.pop("_id")
+			document["type"] = document.pop("_type")
+			if("object_type" in document.keys() and index == "clix"):
+				document["object_type"] = str(document["object_type"])
+			if("property_order" in document.keys() and index == "clix"):
+				document["property_order"] = str(document["property_order"])
+			if("object_value" in document.keys()):
+				document["object_value"] = str(document["object_value"])	#for consistent mapping 
+			create_map(document)
+			doc_type = get_document_type(document)
+			print("indexing document %d with id: %s" % (k,document["id"]["$oid"]))
+			es.index(index=index, doc_type=doc_type, id=document["id"]["$oid"], body=document)
+			if "contributors" in document.keys():
+				contributors = document["contributors"]
+				for contributor_id in contributors:
+					es.index(index = author_index, doc_type = contributor_id,id=document["id"]["$oid"], body = document)
+			# print("indexed document %d with id: %s" % (k,document["id"]["$oid"]))
+			k+=1
+
+def get_document_type(document):
+	if document["type"]=="GSystem":
+		if('if_file' in document.keys()):
+			if document["if_file"]["mime_type"] is not None:
+				data = document["if_file"]["mime_type"].split("/")
+				doc_type = data[0]
+			else:
+				doc_type = "NotMedia"
+		else:
+			doc_type = "NotMedia"
+		#doc_type = "GSystem"			
+	elif document["type"]=="Author":
+		doc_type = "Author"
+	elif document["type"]=="GAttribute":
+		doc_type = "GAttribute"
+	elif document["type"]=="GRelation":
+		doc_type = "GRelation"
+	elif document["type"]=="AttributeType":
+		doc_type = "AttributeType"
+	elif document["type"]=="Filehive":
+		doc_type = "Filehive"
+	elif document["type"]=="RelationType":
+		doc_type = "RelationType"
+	elif document["type"]=="Group":
+		doc_type = "Group"
+	else:
```

Comment:
Alter logic for handling such situations:
- Avoid long chain of `if..elif..else`. It affects performance. Increases lines of code, reduces readability of long files/methods.
- Think of something like:
```
if document["type"] in ["Author", "GAttribute", "", ... ]:
    doc_type = document["type"]
else:
    doc_type = "DontCare"
```